### ISSUE TITLE
Menu | Zoom | Keyboard navigation/Screen reader - When menu opens should auto focus on first item

### ISSUE THREAD

**Issue by LaurieSReynolds** [COVID19Tracking/website#1689]
When open, the menu is rendered above the menu trigger element.   This requires the visitor to reverse tab navigation to reach the menu items.  This is not clear to a screen reader user.

Steps to reproduce
1) Zoom to 200+ x - notice menu displays above the trigger
2) Use keyboard to reach the menu items

Expected behavior
When menu is open, the first menu item is focused or the menu opens below the trigger element so tab will focus the first menu item

![image](https://user-images.githubusercontent.com/18370991/102684744-295d2b80-4190-11eb-919d-308b9d2bfdb2.png)

**Comment by chonghorizons** [comment]
I verified the @LaurieSReynolds issue report. Also, there is another accessibility issue (see below)

TLDR version:  Maybe this can be ignored, since the links can still be reached.

## Possible fix
There is no easy way to fix this without possibly creating another problem. The fix is to could set the `tabindex` attribute:

* COVID TRACKING PROJECT (logo) and MENU/CLOSE is set to `tabindex=1`
* Search and the 5 items above (in the appearing menu) are set to `tabindex=2`
  * When the box is hidden, the these will be skipped 
* Everything else is set to `tabindex=3`

However, this website (https://webaim.org/techniques/keyboard/tabindex) recommends against setting the tab index.
I'm not sure how the framework is setup, but it might be onerous to set he 

## Problem created:
The problem created is that the expected tab behavior no longer goes from top to bottom. (as discussed at https://webaim.org/techniques/keyboard/tabindex)

I do not use tabbing so I'm not sure what would be more frustrating in general for people who do. If the tab order jumps back up, that might confuse some people since it breaks the top-down order. I'm not sure which confusion is more: 
* that the expanding box is above and one needs to use SHIFT+TAB
* or that tab makes it jump up when people might have been expecting it to jump down.

Recommend: no change

## Another accessibility issue, hidden tabbing
When the screen is zoomed in, tabbing is invisible for the elements OurData, About The Data, etc. What I mean: you hit tab 5 times until something shows up. Even though the items aren't rendered (because `opacity:0`), the block is still being rendered.

I find that this is fixed (in inspector tools) by adding the style attribute `display: none` when the menu is hidden.
```<div class="" style="height: 0px; opacity: 0; display: none; overflow: hidden; transition: height 500ms ease-in-out 0s, opacity 500ms ease-in-out 0s, background 500ms ease-in-out 0s;"><div class="b8abe" style="min-height:0px"><div class="_2a9b5"><div class="_08af2"><label ```
When menu is pressed, the code should remove `display: none`.

Without `display: none`, it just is rendered as a zero area box, but the links in the box are still in the rendered DOM, so they get a stop in the tab order.

**Comment by LaurieSReynolds** [comment]
Good find @chonghorizons!

**Comment by LaurieSReynolds** [comment]
@chonghorizons - Agree - for. a person without a vision impairment using the keyboard, it isn't as big a challenge.   A person using a screen reader might be confused as they wouldn't know that they have to backtrack to get to the menu elements.  (I've updated the issue title)

I agree, setting tabIndex doesn't seem like the optimal solution.

Is it not possible to auto focus to the top element when the menu is opened?

Another alternative would be to explore a more traditional menu that shows as an overlay to the content.

**Comment by chonghorizons** [comment]
@LaurieSReynolds that's a good idea. There is probably a way to add a change to refocus the tab focus to the top.

The change might not be easy. I estimate 5 to 10 hours of dev time. The reason is that the menu system is tightly integrated into the Gatsby framework and would probably involve tracing a lot of things.

Maybe this is a low priority, given that the gain may be small and the fix is not fast.

If it's a high priority, an acceptable option may be to have the menu expand downward, retaining the top to bottom.
